modules/core/src/main/java/org/apache/synapse/endpoints/oauth/OAuthUtils.java

```diff
@@ -115,11 +119,31 @@ private static OAuthHandler getSpecificOAuthHandler(OMElement oauthElement) {
         if (clientCredentialsElement != null) {
             oAuthHandler = getClientCredentialsHandler(clientCredentialsElement);
         }
+
+        if (passwordCredentialsElement != null) {
+            oAuthHandler = getPasswordCredentialsHandler(passwordCredentialsElement);
```

Comment:
can we re organize this if conditions, oAuthHandler will be updated only once

Reply:
Fixed with d4bb790